workflow/controller/agent.go

```diff
@@ -128,6 +128,13 @@ func (woc *wfOperationCtx) createAgentPod(ctx context.Context) (*apiv1.Pod, erro
 					Image:           woc.controller.executorImage(),
 					ImagePullPolicy: woc.controller.executorImagePullPolicy(),
 					Env:             envVars,
+
+					VolumeMounts: []apiv1.VolumeMount{{
+						Name:      "cert",
+						MountPath: "/etc/ssl/certs/ca-certificates",
```

Comment:
This file will already exist. So that it why the tests are failing, you're replacing a working file, with an non-working directory.